metrics/sources/summary/summary.go

```diff
@@ -194,10 +194,25 @@ func (this *summaryMetricsSource) decodePodStats(metrics map[string]*MetricSet,
 
 	for _, container := range pod.Containers {
 		key := PodContainerKey(ref.Namespace, ref.Name, container.Name)
+		// This check ensures that we are not replacing metrics of running container with metrics of terminated one if
+		// there are two exactly same containers reported by kubelet.
+		if _, exist := metrics[key]; exist && containerIsTerminated(&container, metrics[key].CreateTime) {
+			continue
```

Comment:
please please please always log conditions like this (at high log level, like 8).  Otherwise, it's impossible to debug user deployments without sending a patched Heapster container to the user.